File: codecheck-case/lua/BKCT_UnreachableCode/BKCT_UnreachableCode_01.lua
function bad_01()
    return -- bad
    print("This code is unreachable")
end

function bad_02()
    if true then
        return -- bad
        print("This code is unreachable")
    end
end

function bad_03()
    for i = 0, 9 do
        if i > 5 then
            break
            -- bad
            local a = 1
        end
    end
end

function bad_04()
    local i = 5
    while i > 0 do
        i = i - 1
        if i < 2 then
            break
            -- bad
            local a = 1
        end
    end
end

File: codecheck-case/cpp/BKCT_UnreachableCode/BKCT_UnreachableCode_01.cpp
#include <cstdio>


void bad_01()
{
    return;
    // bad
    printf("This code is unreachable\n");
}


void bad_02()
{
    if (true)
    {
        return;
        // bad
        int x = 10; // Unreachable
    }
}

void bad_03()
{
    for (int i = 0; i < 10; i++)
    {
        if (i > 5)
        {
            break;
            // bad
            int a = 1;
        }
    }
}

void bad_04()
{
    int i = 5;
    while (i--)
    {
        if (i < 2)
        {
            break;
            // bad
            int a = 1;
        }
    }
}
